lib/components/Table/Table.styled.ts

```diff
@@ -159,7 +161,8 @@ export const TablePerPageSelector = styled(Select)`
   }
 
   ${SelectItems} {
-    top: 48px;
+    top: ${({ autoCompleteItems, expandUp }) =>
+      expandUp ? autoCompleteItems.length * -36 : 48}px;
```

Comment:
this is not a good solution. if the options are multi line or custom components of varying height it will not work.

to make this work properly you must:

- detect if the component is near the window end, and you have to do this with JS using window.height, window.scrollY and a proximity threshold that you have to define (X pixels away from the bottom)
- automatically decide if it should expand up or down (don't leave this to a prop)
- for making it expand up:
 - invert the order of the array of options
 - it's a standard, the first item is the one that is shown CLOSER to the select, not at the top, regardless of direction.
 - you can achieve this using flex inverse properties (flex direction)
 - you move it up using the rendered rect height of the select container and the 'bottom' positioning (eg, if the container is 100px tall, you set the list of items positioning to be bottom: 100px)
 - you can get the actual height of the select input/container using getBoundingClientRect() or a theme variable in case we have one to dictate its height.

Reply:
I will work on that, thanks for putting some thought on this and taking the time to list the suggestions/solution ideas!!